NavfertyExcelAddIn/Commons/EnumerableExtensions.cs

```diff
@@ -49,38 +49,41 @@ public static void ForEachCell(this Range range, Action<Range> action)
                 return;
             }
 
-            // minimize number of COM calls to excel
-            var values = (object[,])rangeValue;
+            if (!(rangeValue is double))
```

Comment:
The private method `ApplyToArea` is used in the static extension method:
```
ApplyForEachCellOfType<TIn, TOut>(this Range range, Func<TIn, TOut> transform) { ... }
```

The main idea is than we can explicitly define `TIn` cell value type - and the transformation will be applied only for the cells, that contain value of such type. According to that, we **should not** check for double type - `TIn` **can** be double in some cases.

I would propose between
```
if (rangeValue is TIn v)
```
and casting
```
var values = (object[,])rangeValue;
```
to insert check that value is neither TIn nor array (in case when single cell is selected and value is different from TIn).

We can reach that with such way by replacing 53 line with [pattern matching](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/is)
```
if (!(rangeValue is object[,] values))
    return;
```

Reply:
If you insert this code between
`if (rangeValue is TIn v)`
and
`var values = (object[,])rangeValue;` (2)
then you need to delete the initialization of the values variable.

I also didn't notice any difference between having this string (2) and not having it.

Reply:
Yes, you are right. The code
```
if (!(rangeValue is object[,] values))
    return;
```
works as initialization of `values` variable.

The point is that we check if the value is not array. It means that selection contains neither single cell with value of target type (`TIn`), nor multiple cells that should be further handled. In that case we just return from method without any changes.

For example, if `range.ApplyForEachCellOfType<string, object>(x => { ... })` was invoked, it means that transformation logic must be applied:

1. If single cell with string value is selected, applied to this cell
1. If single cell with numeric or date value is selected, **not applied** to this cell
1. If multiple cells are selected, applied only for cells with string

The current bug is that we do not handle second scenario and trying to cast cell value (ex. double) to array.